l10n_br_fiscal/models/document.py

```diff
@@ -314,6 +317,28 @@ def create(self, values):
             values["document_date"] = self._date_server_format()
         return super().create(values)
 
+    def unlink(self):
+        forbidden_states_unlink = [
+            SITUACAO_EDOC_AUTORIZADA,
+            SITUACAO_EDOC_CANCELADA,
+            SITUACAO_EDOC_DENEGADA,
+            SITUACAO_EDOC_INUTILIZADA,
+        ]
+
+        for record in self.filtered(
+            lambda d: d != self.env.user.company_id.fiscal_dummy_id
+            and d.edoc_state in forbidden_states_unlink
+        ):
```

Comment:
É meio deselegante fazer um for com um raise dentro assim. Que tal escrever algo como if any(record for record in self.filtered(...):

Reply:
Eu pensei em usar o if any(.... mas eu acabei colocando esse for para quando tiver gerar o erro ser possível imprimir na mensagem do erro o número e o status do documento que gerou u erro.

Reply:
OK, pode ser eu nâo tinha visto.